src/pkg/egress/syslog/https_batch.go

```diff
@@ -110,39 +133,62 @@ func NewHTTPSBatchWriter(
 			egressMetric:    egressMetric,
 			syslogConverter: c,
 		},
-		retryer: Retryer{
-			binding: binding,
+		retryer: &Retryer{
+			binding:       binding,
+			retryDuration: defaultBackoff,
+			maxRetries:    defaultMaxRetries,
 		},
-		batchSize:    256 * 1024,        // Default value
-		sendInterval: 1 * time.Second,   // Default value
-		msgChan:      make(chan []byte), // Buffered channel for messages
+		batchSize:    defaultBatchSize,
+		sendInterval: defaultSendInterval,
+		msgChan:      make(chan []byte, defaultMsgChanBuffer),
 		quit:         make(chan struct{}),
 	}
 
 	for _, opt := range options {
 		opt(writer)
 	}
 
+	d := &dispatcher{}
+	for i := 0; i < defaultNumWorkers; i++ {
+		wrk := &worker{
+			id:      i,
+			input:   make(chan batch, 5), // buffered to prevent blocking dispatcher
+			retryer: writer.retryer,
+			writer:  writer,
+		}
+		go wrk.start()
+		d.workers = append(d.workers, wrk)
+	}
+	writer.dispatcher = d
+
 	writer.wg.Add(1)
 	go writer.startSender()
 
 	return writer
 }
 
+// Write converts an Envelope to syslog messages and queues them for batching.
+// If the msgChan is full, messages are dropped with a warning.
 func (w *HTTPSBatchWriter) Write(env *loggregator_v2.Envelope) error {
 	msgs, err := w.syslogConverter.ToRFC5424(env, w.hostname)
 	if err != nil {
-		log.Printf("Failed to parse syslog, dropping message, err: %s", err)
+		log.Printf("Failed to parse syslog, dropping message, err: %v", err)
 		return nil
 	}
 
 	for _, msg := range msgs {
-		w.msgChan <- msg
+		select {
+		case w.msgChan <- msg:
+		default:
+			log.Printf("msgChan full, dropping message")
+			// Consider adding a metric for dropped messages here
+		}
```

Comment:
there should be no queue here, this channel is only to decouple batching correctly.